File: Program's_Contributed_By_Contributors/Java_Programs/Data_structure/BubbleSort.java
package array.sorting;

import java.util.Arrays;

public class BubbleSort {
    
    static void BSort(int[] array) {

        for (int i = 0; i < array.length; i++) {

            for (int j = 0; j < array.length - i; j++) {

                if (array[j] < array[i]) {
                    swap(array, j - 1, j);
                }
            }
        }
    }

    static void swap(int[] array, int first , int second) {
        int temp = array[first];
        array[first] = array[second];
        array[second] = temp;
    }

    public static void main(String[] args) {
        int[] arr = {1,4,3,2,5,6,8,9};
        BSort(arr);
        System.out.println(Arrays.toString(arr));
    }

}

File: Bubble_Sort.cpp
#include<iostream>
using namespace std;

void BubbleSort(int arr[],int size){
	for(int i = 0 ;i<size ; i++){
		for(int j = 0; j<size -1 - i;j++){
			if(arr[j] >arr[j+1]){
				swap(arr[j],arr[j+1]);
			}
		}
	}
	
	
}
void printArray(int arr[], int size){
	for(int i = 0; i <size ; i++){
		cout<<arr[i]<<endl;
	}
}

int main(){
	
	int arr[6] = {23,10,65,55,34,5};
	cout<<"Array before Sorting :";
	printArray(arr,6);
	cout<<"\nArray After Sorting :";
	BubbleSort(arr,6);
	printArray(arr,6);
	
	
	
	
	
	return 0;
}
